base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1212,114 +1209,42 @@ fn fetch_blocks<T: BlockchainBackend>(
-    //     let kernel_cp = fetch_checkpoint(db, MmrTree::Kernel, horizon_height - 1)?;
-    //     let (kernel_hashes, _) = kernel_cp.into_parts();
-    //     let kernels = fetch_kernels(db, kernel_hashes)?;
-    //     for kernel in kernels {
-    //         if kernel.excess_sig == excess_sig {
-    //             return Ok(None);
-    //         }
-    //     }
-    // }
-    // Err(ChainStorageError::ValueNotFound {
-    //     entity: "Kernel".to_string(),
-    //     field: "Excess sig".to_string(),
-    //     value: excess_sig.get_signature().to_hex(),
-    // })
-}
-
-fn fetch_block_with_utxo<T: BlockchainBackend>(
-    _db: &T,
-    _commitment: Commitment,
+    db: &T,
+    excess_sig: Signature,
 ) -> Result<Option<HistoricalBlock>, ChainStorageError>
 {
-    unimplemented!()
-    // let metadata = db.fetch_chain_metadata()?;
-    // let db_height = metadata.height_of_longest_chain.unwrap_or(0);
-    // let horizon_height = metadata.horizon_block(db_height);
-    // for i in (horizon_height..db_height).rev() {
-    //     let utxo_cp = fetch_checkpoint(db, MmrTree::Utxo, i)?;
-    //     let (utxo_hashes, _) = utxo_cp.into_parts();
-    //     let utxos = fetch_outputs(db, utxo_hashes)?;
-    //     for utxo in utxos.0 {
-    //         if utxo.commitment == commitment {
-    //             return Ok(Some(fetch_block(db, i)?));
-    //         }
-    //     }
-    //     for comm in utxos.1 {
-    //         if comm == commitment {
-    //             return Ok(Some(fetch_block(db, i)?));
-    //         }
-    //     }
-    // }
-    // // data is not in the pruning horizon, let's check behind that but only if there is a pruning horizon
-    // if horizon_height > 0 {
-    //     let utxo_cp = fetch_checkpoint(db, MmrTree::Utxo, horizon_height - 1)?;
-    //     let (utxo_hashes, _) = utxo_cp.into_parts();
-    //     let utxos = fetch_outputs(db, utxo_hashes)?;
-    //     for utxo in utxos.0 {
-    //         if utxo.commitment == commitment {
-    //             return Ok(None);
-    //         }
-    //     }
-    //     for comm in utxos.1 {
-    //         if comm == commitment {
-    //             return Ok(None);
-    //         }
-    //     }
-    // }
-    // Err(ChainStorageError::ValueNotFound {
-    //     entity: "Utxo".to_string(),
-    //     field: "Commitment".to_string(),
-    //     value: commitment.to_hex(),
-    // })
+    match db.fetch_kernel_by_excess_sig(&excess_sig) {
+        Ok(kernel) => match kernel {
+            Some((_kernel, hash)) => fetch_block_by_hash(db, hash),
+            None => Ok(None),
+        },
+        Err(_) => Err(ChainStorageError::ValueNotFound {
+            entity: "Kernel".to_string(),
+            field: "Excess sig".to_string(),
+            value: excess_sig.get_signature().to_hex(),
+        }),
+    }
 }
 
-fn fetch_block_with_stxo<T: BlockchainBackend>(
-    _db: &T,
-    _commitment: Commitment,
+fn fetch_block_with_transaction_output<T: BlockchainBackend>(
```

Comment:
This function only fetches the block the UTXO was created in, not spent

Reply:
Thanks for clarifying, will fix.